File: fix_menu_tags.py

import os
import re

def fix_menu_toggle():
    for filename in os.listdir('.'):
        if filename.endswith('.html'):
            with open(filename, 'r', encoding='utf-8') as f:
                content = f.read()
            
            # Pattern: <button class="menu-toggle"...> ... </div>
            # We want to replace the FIRST </div> after the button start with </button>
            # But only if it hasn't been replaced already.
            
            # Find all occurrences of the button start
            matches = list(re.finditer(r'<button class="menu-toggle"[^>]*>', content))
            
            if not matches:
                continue
                
            new_content = content
            # Process in reverse order to not mess up indices
            for match in reversed(matches):
                start_index = match.end()
                # Find the next </div>
                next_div = content.find('</div>', start_index)
                if next_div != -1:
                    # Check if there are any other opening divs between match and next_div
                    # There generally shouldn't be for <i class="..."></i>
                    # But let's be safe.
                    chunk = content[start_index:next_div]
                    if '<div' not in chunk:
                        # Safe to replace
                        new_content = new_content[:next_div] + '</button>' + new_content[next_div+6:]
            
            if new_content != content:
                print(f"Fixing {filename}")
                with open(filename, 'w', encoding='utf-8') as f:
                    f.write(new_content)

if __name__ == '__main__':
    fix_menu_toggle()

File: fix_menu_tags.js

const fs = require('fs');
const path = require('path');

const directoryPath = '.';

function fixMenuToggle() {
    fs.readdir(directoryPath, (err, files) => {
        if (err) {
            console.log('Unable to scan directory: ' + err);
            return;
        }

        files.forEach((file) => {
            if (path.extname(file) === '.html') {
                const filePath = path.join(directoryPath, file);
                fs.readFile(filePath, 'utf8', (err, data) => {
                    if (err) {
                        console.log(err);
                        return;
                    }

                    // We need to replace the FIRST </div> after <button class="menu-toggle"...>
                    // BUT ONLY IF it hasn't been replaced.
                    // The problematic button looks like:
                    // <button class="menu-toggle" ...>
                    //    <i ...></i>
                    // </div>

                    // Regex match the opening button tag
                    const regex = /<button class="menu-toggle"[^>]*>([\s\S]*?)<\/div>/g;

                    // Replace with </button> 
                    // Capture group 1 is the content inside (icons etc)
                    const newData = data.replace(regex, (match, p1) => {
                        // Ensure there are no nested divs inside p1 that would make this match incorrect
                        if (p1.includes('<div')) {
                            // If there's a div inside, regex might have matched too far or correct.
                            // But for menu-toggle which just has <i> usually, it's fine.
                            // However, regex is greedy/lazy. [\s\S]*? is lazy.
                            // So it stops at the first </div>.
                            // If the content has a div, we might be breaking it.
                            // But menu-toggle content is: <i class="fas fa-bars"></i>
                            return match; // Don't touch if complex
                        }
                        return match.replace('</div>', '</button>');
                    });

                    if (newData !== data) {
                        fs.writeFile(filePath, newData, 'utf8', (err) => {
                            if (err) console.log(err);
                            else console.log(`Fixed ${file}`);
                        });
                    }
                });
            }
        });
    });
}

fixMenuToggle();
